lib/config.ml

```diff
@@ -31,27 +30,56 @@ module DN = struct
   let sexp_of_t _ = Sexplib.Sexp.Atom "distinguished name"
 end
 
+type ticket_cache = {
+  lookup : Cstruct.t -> (psk13 * epoch_data) option ;
+  ticket_granted : psk13 -> epoch_data -> unit ;
+  lifetime : int32 ;
+  timestamp : unit -> Ptime.t
+}
```

Comment:
It seems to me this structure heavily suggests using mutable state (with callbacks taking care of the mutation). Would it make sense to support a monadic approach like sketched below (where the mutable versions can just have `unit ticket_cache`?
```ocaml
type 'a ticket_cache = {
  cache_state : 'a ;
  lookup : 'a -> Cstruct.t -> (psk13 * epoch_data) option * 'a ;
  ticket_granted : 'a -> psk13 -> epoch_data -> 'a ;
  lifetime : int32 ; (* TTL seconds or absolute timestamp ? *)
  timestamp : 'a -> unit -> Ptime.t ;
}
(* alternatively I guess [ticket_cache] could be a module signature and the user would have to provide
    a module. *)
```
I'm not sure what your policy is, but I think it would be nice to eventually be able to serialize/deserialize the entire TLS state. Both for debugging/tracing/introspection and for applications that need to transfer it (supporting for instance session hand-off). I know that it's not immediately possible at the moment, but what do you think about that as an eventual goal?

Reply:
I prefer feature requests as separate issues.